fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java

```diff
@@ -215,6 +215,9 @@ public static Optional<DataType> findTightestCommonType(BinaryOperator binaryOpe
             }
         } else if (left instanceof CharacterType && right instanceof CharacterType) {
             tightestCommonType = CharacterType.widerCharacterType((CharacterType) left, (CharacterType) right);
+        } else if (left instanceof CharacterType && right instanceof DateLikeType) {
+            // TODO: need check implicitCastMap to keep the behavior consistent with old optimizer
+            tightestCommonType = right;
```

Comment:
should we also add `right instanceof CharacterType && left instanceof DateLikeType`